### PR TITLE
fix(dot/babe): use `bs.latestFinalised` instead of using `round/set id`

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- This change fixes the problem where:
  - While syncing in the Westend chain Gossamer stores the finalized block hash bounding it to its respective round and set id, but in Westend happened updates to validators where they restarted the finalization in the same set id but reset the round counter. As described in https://github.com/ChainSafe/gossamer/issues/3150, we synced block 198,656 with round 106 and set 314 and then we synced block 198,785 with round *3* and set id 314
  - The fact that a higher block contains a lower round for the same set id was ignored by Gossamer, which means that Gossamer finalizes block 198,785 but did not bound the hash with the round and set id
  - When the next synced block arrives to be verified and finalized Gossamer understands that the latest finalized block is 198,656 and the `RangeInMemory` method fails to find that block since it is already pruned from memory.

- In `handleFinalisedBlock` use `bs.latestFinalised` instead of using `round/set id`
- Remove from `setHighestRoundAndSetID` the verification of the given round and set id compared to the current round and set id. 

## Tests

After running a Gossamer node for 10+ hours it is finalising blocks keeping the amount of tries in memory low

image

## Issues

- Closes https://github.com/ChainSafe/gossamer/issues/3150

## Primary Reviewer

@jimjbrettj

### CODE REVIEW

**Comment by EclesioMeloJunior** [comment]
This strict validation was removed since in Westend there are a lot of weird cases that lead to rounds being messed up in the same set id, but I still wondering about adding an option called `validation-while-syncing` with 2 modes `relaxed` and `strict` than we can make this validation strict while syncing against Kusama/Polkadot and make it relaxed while syncing against Westend. @danforbes

**Review by kishansagathiya — APPROVED** [review]

**Review by dimartiro — APPROVED** [review]

**Comment by jimjbrettj** [comment]
It seems that in state tests in block_finalization_test.go that TestHighestRoundAndSetID is failing

File: dot/state/block_finalisation.go [hunk 1]
```diff
@@ -63,16 +63,6 @@ func (bs *BlockState) GetFinalisedHash(round, setID uint64) (common.Hash, error)
 }
 
 func (bs *BlockState) setHighestRoundAndSetID(round, setID uint64) error {
-	currRound, currSetID, err := bs.GetHighestRoundAndSetID()
-	if err != nil {
-		return err
-	}
-
-	// higher setID takes precedence over round
-	if setID < currSetID || setID == currSetID && round <= currRound {
```
Comment (jimjbrettj) [inline dot/state/block_finalisation.go:72]: I know that for relaxed mode we don't need to verify the round number, but do we also now skip verifying that the setId is greater?

**Review by jimjbrettj — CHANGES_REQUESTED** [review]
Looks good, just need to fix failing tests

**Review by jimjbrettj — APPROVED** [review]
Nice work!

**Pull request merged**